apps/gdalalg_vector_geom_simplify.cpp

```diff
@@ -30,24 +30,11 @@ GDALVectorGeomSimplifyAlgorithm::GDALVectorGeomSimplifyAlgorithm(
     : GDALVectorGeomAbstractAlgorithm(NAME, DESCRIPTION, HELP_URL,
                                       standaloneStep, m_opts)
 {
-    auto &arg =
-        AddArg("tolerance", 0, _("Distance tolerance for simplification."),
-               &m_opts.m_tolerance)
-            .SetPositional()
-            .SetRequired();
-    arg.AddValidationAction(
-        [this, &arg]()
-        {
-            const double val = arg.Get<double>();
-            if (!(val > 0))
-            {
-                ReportError(
-                    CE_Failure, CPLE_IllegalArg,
-                    "Value of 'tolerance' should be strictly positive.");
-                return false;
-            }
-            return true;
-        });
+    AddArg("tolerance", 0, _("Distance tolerance for simplification."),
+           &m_opts.m_tolerance)
+        .SetPositional()
+        .SetRequired()
+        .SetMinValueExcluded(0);
```

Comment:
ST_Simplify(0) in PostGIS and corresponding things is other software is a common trick for fixing some sort of geometry problems like repeated points. There may be better options nowadays but is there any reason not to allow `.SetMinValueIncluded(0)`? On the other hand, the value must have been strictly positive even before.

Reply:
good point. Changed to SetMinValueIncluded(0)